src/connectors.rs

```diff
@@ -1100,6 +1101,14 @@ pub(crate) trait Connector: Send {
 
     /// Returns the codec requirements for the connector
     fn codec_requirements(&self) -> CodecReq;
+
+    fn validate(&self, config: &ConnectorConfig) -> Result<()> {
```

Comment:
instead of a "supports" / "not supports" could we use the same tactic we use for dispatching a command and instead of executing it returning valid / invalid? So parse the command into a struct, check if the struct is valid, check if the connector in question support this type of command.

Not sure on the implementation cost of this, but it's worth a thought as it'd be more helpful then a global supported / not supported.

So basically a hygenic error that validates the command during compile time to the limits that are possible (as in not checking of the file is readable, but checking if the connector supports file IO commands and possibly letting it check if the connector supports this specific command.)

Reply:
I think we can do it with the current setup - the command can be matched upon in the validate method, and then the method can decide if it passes

Reply:
(it is passed as an unserialized enum to ConnectorConfig)